### ISSUE TITLE
Queries are sometimes sent from all-zeroes source MAC address

### ISSUE THREAD

**Issue by wkz** [kernelkit/mcd#4]
When starting `mcd` on a newly created interface, the MAC address can not always be determined. When that happens, `mcd` soldiers on and uses an SA of all zeroes.

Preferably, we would not allow any queries to be sent on an interface until a proper SA has been selected.

**Comment by troglobit** [comment]
Indeed, with no SA mcd should go back to sleep and wake up on the next startup-query-interval, see #5.

**Comment by wkz** [comment]
I looked a bit at the source. Making some notes here.

It seems like we only get one shot, one opportunity, to capture the interface SA - and we unfortunately let it slip:

https://github.com/kernelkit/mcd/blob/23cba467298287a5e00563ad25e0cd634dca07af/src/config.c#L133-L134

This is the only caller of `getmac()` AFAIK, so it seems like if the config parser runs before the interface exists, then we're never going to fill in the SA.

Unless we want to do some fancy netlink parsing, I think we need to call `getmac()` on every `RTM_{NEW,SET}LINK` to make sure we keep `mcd` in sync with the interface config.

While we're there I think we need to consider the condition for starting the interface. At the moment, we settle for `IFF_UP`, i.e. admin up:

https://github.com/kernelkit/mcd/blob/23cba467298287a5e00563ad25e0cd634dca07af/src/iface.c#L249-L252

Should we maybe also require that the link is oper up, i.e. `IFF_RUNNING`?

If so, that might require us to handle `RTM_SETLINK` notifications as well:

https://github.com/kernelkit/mcd/blob/23cba467298287a5e00563ad25e0cd634dca07af/src/netlink.c#L48-L58
